### ISSUE TITLE
Lua filter: attr accessor can not be used in conjunction with classes accessor

### ISSUE THREAD

**Issue by zkamvar** [jgm/pandoc#7843]
**Explain the problem.**

I came across an issue that was working in Pandoc 2.11--2.15, but failing in the nightly builds, If I want to create a lua filter that nests divs, I must set the `classes` element before I set the `attr` element, or I will get the following error:

```
Error running filter attr-class.lua:
PandocLuaError "string expected, got table\n\twhile retrieving value\n\twhile retrieving key-value pair\n\twhile retrieving HTML-like attributes\n\twhile retrieving Attr"
stack traceback:
	[C]: in function 'pandoc.Div'
	attr-class.lua:5: in function 'Div'
```

I now have a new problem: _if I set the classes element before I set the `attr` element, then the classes disappear_.

**Solution/Workaround**

I have found that if I specify the class along with the id in the attr setter, then it will work, though if I have variable classes, I will have to do string replacement/concatenation instead of table insertion to achieve the same effect. 

**MWE**

I have prepared a minimal working example to reproduce the error:

attr-class.lua

```lua
function Div(el) 
  local div = pandoc.Div({el})
  div.attr = {["id"] = "one"}
  div.classes = {"one"}
  local vid = pandoc.Div({div})
  vid.attr = {["id"] = "two"}
  vid.classes = {"two"}
  return(vid)
end
```

class-attr.lua

```lua
function Div(el) 
  local div = pandoc.Div({el})
  div.classes = {"one"}
  div.attr = {["id"] = "one"}
  local vid = pandoc.Div({div})
  vid.classes = {"two"}
  vid.attr = {["id"] = "two"}
  return(vid)
end
```

attr.lua

```lua
function Div(el) 
  local div = pandoc.Div({el})
  div.attr = {["id"] = "one", ["class"] = "one"}
  local vid = pandoc.Div({div})
  vid.attr = {["id"] = "two", ["class"] = "two"}
  return(vid)
end
```

input.md

```markdown
::: test
this is a test
:::
```

```bash
$ docker run --rm --volume "`pwd`:/data" --user `id -u`:`id -g` pandoc/core -t html -f markdown --section-divs --lua-filter attr-class.lua test.md
Error running filter attr-class.lua:
PandocLuaError "string expected, got table\n\twhile retrieving value\n\twhile retrieving key-value pair\n\twhile retrieving HTML-like attributes\n\twhile retrieving Attr"
stack traceback:
	[C]: in function 'pandoc.Div'
	attr-class.lua:5: in function 'Div'
```
```bash
$ docker run --rm --volume "`pwd`:/data" --user `id -u`:`id -g` pandoc/core -t html -f markdown --section-divs --lua-filter class-attr.lua test.md
<div id="two">
<div id="one">
<div class="test">
<p>this is a test</p>
</div>
</div>
</div>
```

```bash
$ docker run --rm --volume "`pwd`:/data" --user `id -u`:`id -g` pandoc/core -t html -f markdown --section-divs --lua-filter attr.lua test.md
<div id="two" class="two">
<div id="one" class="one">
<div class="test">
<p>this is a test</p>
</div>
</div>
</div>
```

**Pandoc version?**

Pandoc 2.17 (pandoc/core:latest on Docker)

**Comment by tarleb** [comment]
Thanks for the report. I don't see the behavior as a bug, let me explain:

The `attr` can be either a triple containing the id, classes, and attributes, or it can be an HTML-like table. When setting the `div.attr = {id = 'one'}`, then it becomes an HTML-like table: all attributes must be strings, classes must be given as a single space-separated list below key `class`.

The alternative is a triple `{'identifier', {'cls1', 'cls2'}, {{'key', 'value'}}}`, which is close to pandoc's internal representation.

The value `div.classes` is an alias for `div.attr[2]`, and `div.identifier` is an alias for `div.attr[1]`. So setting `div.classes = {'one'}`, is equivalent to defining `attr = {id = 'one', [2] = {'one'}}`. That's neither a triple nor a HTML-like attributes table, so pandoc gets confused.

Now the reason that this used to work is that `Attr` values used to be "smarter", in that they tried harder to make things work. We had to stop doing that for technical reasons.

In conclusion: use `div.identifier = 'one'` to assign an ID and things should work.

**Comment by zkamvar** [comment]
It is as I suspected: not a bug, making the filter API a bit more clear. So, if I understand correctly, the situation is this:

1. If I want to set classes, I pass a table to `div.classes`
2. If I want to set the id, I pass a string to `div.identifier`
3. If I want to set the attributes, I pass a dictionary to `div.attributes`

So if I were to create a div tag that would contain an identifier, classes, and attributes, it should look something like this: 

```lua
function Div(el)
  local div = pandoc.Div({el})
  div.identifier = 'one'
  div.classes = {'oneClass', 'anotherClass'}
  div.attributes = {
    ['data-bs-parent'] = 'two',
    ['aria-labelledby'] = 'two'
  }
  return(div)
end
```

**Comment by tarleb** [comment]
Yes, exactly.

Just for completeness: The above simplified a little, `div.attr` is not really a triple but a Lua "userdata" object of type `Attr`. It has a few additional properties, like yielding the Haskell representation when passed to `tostring`. But that's a detail that you'll probably don't need, we accept triples and HTML-like attribute tables in it's place.
